src/app/pages/courses-page/components/course/course.component.ts

```diff
@@ -7,6 +7,7 @@ import { DeleteCourseModalComponent } from './delete-course-modal/delete-course-
   selector: 'app-course',
   templateUrl: './course.component.html',
   styleUrls: ['./course.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CourseComponent implements OnInit{
```

Comment:
Opening of deleting modal should be in smart component. You need to use event binding here.